heft-plugins/heft-sass-plugin/src/SassTypingsGenerator.ts

```diff
@@ -95,25 +110,53 @@ export class SassTypingsGenerator extends StringValuesTypingsGenerator {
       fileExtensions,
       filesToIgnore: sassConfiguration.excludeFiles,
 
+      getAdditionalOutputFiles: getCssPaths,
+
       // Generate typings function
-      parseAndGenerateTypings: async (fileContents: string, filePath: string) => {
+      parseAndGenerateTypings: async (fileContents: string, filePath: string, relativePath: string) => {
         if (this._isSassPartial(filePath)) {
           // Do not generate typings for Sass partials.
           return;
         }
+
         const css: string = await this._transpileSassAsync(
           fileContents,
           filePath,
           buildFolder,
           sassConfiguration.importIncludePaths
         );
-        const classNames: string[] = await this._getClassNamesFromCSSAsync(css, filePath);
-        const sortedClassNames: string[] = classNames.sort((a, b) => a.localeCompare(b));
-        const sassTypings: IStringValueTypings = { typings: [] };
-        for (const exportName of sortedClassNames) {
-          sassTypings.typings.push({ exportName });
+
+        let classMap: IClassMap = {};
+        const cssModulesClassMapPlugin: postcss.Plugin<TCssModules> = cssModules({
+          getJSON: (cssFileName: string, json: IClassMap) => {
+            classMap = json;
+          },
+          // Avoid unnecessary name hashing.
+          generateScopedName: (name: string) => name
+        });
+
+        await postcss([cssModulesClassMapPlugin]).process(css, { from: filePath });
+
+        if (getCssPaths) {
+          await Promise.all(
+            getCssPaths(relativePath).map((cssFile: string) =>
+              FileSystem.writeFileAsync(cssFile, css, {
+                ensureFolderExists: true
+              })
+            )
+          );
```

Comment:
```suggestion
          await Promise.all(
            getCssPaths(relativePath).map(async (cssFile: string) =>
              await FileSystem.writeFileAsync(cssFile, css, {
                ensureFolderExists: true
              })
            )
          );
```

So this shows up in error stacks.

Should this use `Async.forEachAsync`?

Reply:
The TypingsGenerator currently runs on all files serially, so at most this will have the concurrency of the number of requested output folders.